dts/bindings/can/microchip,mcp2515.yaml

```diff
@@ -18,3 +18,6 @@ properties:
     int-gpios:
       type: compound
       category: required
+    reg:
+      type: array
```

Comment:
Sorry for my ignorance, what does `reg:` and `type: array` mean hear for the binding?

Reply:
`reg:` is the property name expected in the dts.  `array` says its an integer list of items.

Reply:
It doesn't really mean anything for the mcp2515, usually it would be a memory address or the i2c bus address, in this case it's from the spi-device.yaml. The 'reg' propery was already in the overlay dts file, but now I had to add it to yaml, otherwise the dts parser assumed the property was modified from required to optional. I couldn't figure out why, but this small change fixed that.

Reply:
Ok cheers, so currently it's a requirement that is fulfilled with the <0> in the overlay.

Reply:
@galak dealt with the matter here in a cleaner way
https://github.com/zephyrproject-rtos/zephyr/pull/18191